### ISSUE TITLE
line_ending is incorrect for &str parser with multi-byte UTF-8 character

### ISSUE THREAD

**Issue by rroelke** [Geal/nom#655]
Reproducer:
```
      use nom::{line_ending, not_line_ending};
      named!(twolines(&str) -> (&str, &str),
        do_parse!(l1 : not_line_ending >>
                  line_ending >>
                  l2 : not_line_ending >>
                  line_ending >>
                  ((l1, l2))));

      assert_eq!(twolines("foo\nbar\n"), Done("", ("foo", "bar")));
      assert_eq!(twolines("féo\nbar\n"), Done("", ("féo", "bar")));
```

Running this results in:
```
	thread 'str::test::utf8_indexing' panicked at 'assertion failed: `(left == right)`
  left: `Error(CrLf)`,
 right: `Done("", ("féo", "bar"))`', str.rs:655:6
```

**Comment by rroelke** [comment]
```not_line_ending``` correctly identifies the index of the newline character and passes it to ```&str::slice``` which then returns the incorrect result.

**Comment by Geal** [comment]
Thanks for the report, I'll look into it

**Comment by Geal** [comment]
ok, this is a really good bug, and I'll have to comb through a lot of code to fix all the instances of this issue :)

**Comment by Geal** [comment]
this should be fixed by 7bc5d42, now looking into the rest of the code to see if it happens anywhere else

**Comment by rroelke** [comment]
Thanks for the quick turnaround!
